engine/table/src/main/java/io/deephaven/engine/table/impl/lang/QueryLanguageParser.java

```diff
@@ -1021,20 +1021,22 @@ public static boolean isTypedVector(Class<?> type) {
     @Override
     public Class<?> visit(BinaryExpr n, VisitArgs printer) {
         BinaryExpr.Operator op = n.getOperator();
+        final Expression leftExpr = n.getLeft();
+        final Expression rightExpr = n.getRight();
 
-        Class<?> lhType = getTypeWithCaching(n.getLeft());
-        Class<?> rhType = getTypeWithCaching(n.getRight());
+        Class<?> lhType = getTypeWithCaching(leftExpr);
+        Class<?> rhType = getTypeWithCaching(rightExpr);
 
         if ((lhType == String.class || rhType == String.class) && op == BinaryExpr.Operator.PLUS) {
 
             if (printer.hasStringBuilder()) {
-                n.getLeft().accept(this, printer);
+                leftExpr.accept(this, printer);
             }
 
             printer.append(getOperatorSymbol(op));
 
             if (printer.hasStringBuilder()) {
-                n.getRight().accept(this, printer);
+                rightExpr.accept(this, printer);
```

Comment:
"Since you're here", this code could be simplified somewhat to

```
            if (printer.hasStringBuilder()) {
                leftExpr.accept(this, printer);
                printer.append(getOperatorSymbol(op));
                rightExpr.accept(this, printer);
            }
```